db/sql/00_msar.sql

```diff
@@ -834,95 +842,71 @@ $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 ----------------------------------------------------------------------------------------------------
 ----------------------------------------------------------------------------------------------------
 
-
--- Rename schema -----------------------------------------------------------------------------------
-
-CREATE OR REPLACE FUNCTION
-__msar.rename_schema(old_sch_name text, new_sch_name text) RETURNS TEXT AS $$/*
-Change a schema's name, returning the command executed.
+DROP FUNCTION IF EXISTS msar.rename_schema(oid, text);
+CREATE OR REPLACE FUNCTION msar.rename_schema(sch_id oid, new_sch_name text) RETURNS void AS $$/*
+Change a schema's name
 
 Args:
-  old_sch_name: A properly quoted original schema name
-  new_sch_name: A properly quoted new schema name
+  sch_id: The OID of the schema to rename
+  new_sch_name: A new for the schema, UNQUOTED
 */
 DECLARE
-  cmd_template text;
-BEGIN
-  cmd_template := 'ALTER SCHEMA %s RENAME TO %s';
-  RETURN __msar.exec_ddl(cmd_template, old_sch_name, new_sch_name);
-END;
-$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
-
-
-CREATE OR REPLACE FUNCTION
-msar.rename_schema(old_sch_name text, new_sch_name text) RETURNS TEXT AS $$/*
-Change a schema's name, returning the command executed.
-
-Args:
-  old_sch_name: An unquoted original schema name
-  new_sch_name: An unquoted new schema name
-*/
-BEGIN
-  RETURN __msar.rename_schema(quote_ident(old_sch_name), quote_ident(new_sch_name));
-END;
-$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
-
-
-CREATE OR REPLACE FUNCTION msar.rename_schema(sch_id oid, new_sch_name text) RETURNS TEXT AS $$/*
-Change a schema's name, returning the command executed.
-
-Args:
-  sch_id: The OID of the original schema
-  new_sch_name: An unquoted new schema name
-*/
+  old_sch_name text := msar.get_schema_name(sch_id);
 BEGIN
-  RETURN __msar.rename_schema(__msar.get_schema_name(sch_id), quote_ident(new_sch_name));
+  IF old_sch_name = new_sch_name THEN
+    -- Return early if the names are the same. This avoids an error from Postgres.
+    RETURN;
+  END IF;
+  EXECUTE format('ALTER SCHEMA %I RENAME TO %I', old_sch_name, new_sch_name);
 END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
--- Comment on schema -------------------------------------------------------------------------------
+CREATE OR REPLACE FUNCTION msar.set_schema_description(
+  sch_id oid,
+  description text
+) RETURNS void AS $$/*
+Set the PostgreSQL description (aka COMMENT) of a schema.
 
-CREATE OR REPLACE FUNCTION
-__msar.comment_on_schema(sch_name text, comment_ text) RETURNS TEXT AS $$/*
-Change the description of a schema, returning command executed.
+Descriptions are removed by passing an empty string. Passing a NULL description will cause
```

Comment:
Huh. Apparently `NULL` and empty string descriptions are equivalent as far as PostgreSQL is concerned. I didn't know that.